src/js/node/stream.ts

```diff
@@ -4350,7 +4354,85 @@ var require_writable = __commonJS({
     Writable.prototype[EE.captureRejectionSymbol] = function (err) {
       this.destroy(err);
     };
-    var webStreamsAdapters;
+    var webStreamsAdapters = {
+      newStreamWritableFromWritableStream(
+        writableStream: WritableStream,
+        options: {
+          highWaterMark?: number;
+          encoding?: string;
+          objectMode?: boolean;
+          signal?: AbortSignal;
+        } = {},
+      ) {
+        if (!isWritableStream(writableStream)) {
+          throw new ERR_INVALID_ARG_TYPE("writableStream", "WritableStream", writableStream);
+        }
+
+        validateObject(options, "options");
+        const {
+          highWaterMark,
+          encoding,
+          objectMode = false,
+          signal,
+          // native = true,
+        } = options;
+
+        if (encoding !== undefined && !Buffer.isEncoding(encoding)) {
+          // TODO: why does this err in typescript?
+          throw new ERR_INVALID_ARG_VALUE(encoding, "options.encoding");
+        }
+        validateBoolean(objectMode, "options.objectMode");
+
+        const nativeStream = getNativeWritableStream(writableStream, options);
```

Comment:
What happens if it's not a `NativeWritable`? Such as if it has no `$bunNativePtr` and the developer did `new WritableStream` and then passed that to this function?